File: web/src/utils/dateUtils.js
export const formatDate = (date, format = 'dd-MM-yyyy') => {
  // Handle invalid dates
  if (!date || isNaN(new Date(date).getTime())) {
    return 'Invalid date'
  }
  
  const d = new Date(date)
  
  // Double check date is valid
  if (isNaN(d.getTime())) {
    return 'Invalid date'
  }
  
  const day = String(d.getDate()).padStart(2, '0')
  const month = String(d.getMonth() + 1).padStart(2, '0')
  const year = d.getFullYear()
  const monthNames = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
  
  return format
    .replace('dd', day)
    .replace('MMM', monthNames[d.getMonth()])
    .replace('MM', month)
    .replace('yyyy', year)
}

export const getDateFromMillis = (millis) => {
  return new Date(parseInt(millis))
}

export const getMillisFromDate = (date) => {
  return date.getTime()
}

export const getStartOfDay = (date) => {
  const d = new Date(date)
  d.setHours(0, 0, 0, 0)
  return d
}

export const getEndOfDay = (date) => {
  const d = new Date(date)
  d.setHours(23, 59, 59, 999)
  return d
}

File: mobile-app/mobile-flutter-poc/lib/utils/date_utils.dart
import 'package:intl/intl.dart';

class AppDateUtils {
  static String formatDate(DateTime date, {String format = 'dd-MM-yyyy'}) {
    if (date.isBefore(DateTime(1900))) {
      return 'Invalid date';
    }

    final day = date.day.toString().padLeft(2, '0');
    final month = date.month.toString().padLeft(2, '0');
    final year = date.year.toString();

    switch (format) {
      case 'dd MMM yyyy':
        return DateFormat('dd MMM yyyy').format(date);
      case 'dd MMMM yyyy':
        return DateFormat('dd MMMM yyyy').format(date);
      case 'dd-MM-yyyy':
        return '$day-$month-$year';
      default:
        return '$day-$month-$year';
    }
  }

  static DateTime getDateFromMillis(int millis) {
    return DateTime.fromMillisecondsSinceEpoch(millis);
  }

  static int getMillisFromDate(DateTime date) {
    return date.millisecondsSinceEpoch;
  }

  static DateTime getStartOfDay(DateTime date) {
    return DateTime(date.year, date.month, date.day);
  }

  static DateTime getEndOfDay(DateTime date) {
    return DateTime(date.year, date.month, date.day, 23, 59, 59, 999);
  }

  static String formatTime(DateTime date) {
    final hours = date.hour.toString().padLeft(2, '0');
    final minutes = date.minute.toString().padLeft(2, '0');
    return '$hours:$minutes';
  }
}

